Software Garden Processing and Aggregate Single Cells

MattsonCam opened the pull request:
This pr software gardens the processing module and supplies the code for aggregating single cell data. Single cell data will be aggregated to the well level for both QC'd and non-QC'd data pre and post feature selection. When you review, I would appreciate any constructive feedback.

Commit:
Added code to aggregate normalized and
feature selected single cell data, and
updated the readme.

Commit:
Refactored the processing script to
normalize and feature select single cell morphology data

Commit:
Updated the README and comments to increase clarity

gwaybio commented on 1.process_data/README.md:
```diff
@@ -1,9 +1,11 @@
-In this module, we perform two preprocessing steps on the SQLite files using [pycytominer](https://github.com/cytomining/pycytominer/tree/main):
+In this module, we perform four preprocessing steps on the SQLite files using [pycytominer](https://github.com/cytomining/pycytominer/tree/main):
 
 1. Merge and annotate single cells from the SQLite file using the [pycytominer SingleCell class](https://github.com/cytomining/pycytominer/blob/main/pycytominer/cyto_utils/cells.py)
```
Do we still do this? (or do we use CytoTable)?

MattsonCam replied:
We use pycytominer. I think we we're planning to switch to cytotable at some point, but decided against it. I'm seeing some cytotable code in `0.download_data` so I think it would be worth asking about this. Also, I think the "JUMP single-cell tasks and deadlines" mentions to process the data with pycytominer

gwaybio commented on 1.process_data/README.md:
```diff
@@ -1,9 +1,11 @@
 
 1. Merge and annotate single cells from the SQLite file using the [pycytominer SingleCell class](https://github.com/cytomining/pycytominer/blob/main/pycytominer/cyto_utils/cells.py)
-2. [Normalize](https://github.com/cytomining/pycytominer/blob/main/pycytominer/normalize.py) the single cells using the negative controls (e.g., DMSO for compound treatment, no-target or target intergenic region sgRNAs for crispr treatment, and genes with weak signatures in orf treatment) as reference for the standard scalar method.
+2. [Normalize](https://github.com/cytomining/pycytominer/blob/main/pycytominer/normalize.py) the single cells using the negative controls (e.g., DMSO for compound treatment, no-target or target intergenic region sgRNAs for crispr treatment, and genes with weak signatures in orf treatment) as reference for the standard scalar method per plate.
```
the "genes with weak signatures" portion - is this what they did in the JUMP paper? We should perform the exact same normalization

MattsonCam replied:
I may reach out to the JUMP group, because I think somebody else included this

MattsonCam replied:
I talked with Jenna, and I this phrase is just to describe the negative controls from here: https://github.com/jump-cellpainting/JUMP-Target

gwaybio commented on 1.process_data/README.md:
```diff
@@ -1,9 +1,11 @@
 1. Merge and annotate single cells from the SQLite file using the [pycytominer SingleCell class](https://github.com/cytomining/pycytominer/blob/main/pycytominer/cyto_utils/cells.py)
-2. [Normalize](https://github.com/cytomining/pycytominer/blob/main/pycytominer/normalize.py) the single cells using the negative controls (e.g., DMSO for compound treatment, no-target or target intergenic region sgRNAs for crispr treatment, and genes with weak signatures in orf treatment) as reference for the standard scalar method.
+2. [Normalize](https://github.com/cytomining/pycytominer/blob/main/pycytominer/normalize.py) the single cells using the negative controls (e.g., DMSO for compound treatment, no-target or target intergenic region sgRNAs for crispr treatment, and genes with weak signatures in orf treatment) as reference for the standard scalar method per plate.
+3. [Feature Select](https://github.com/cytomining/pycytominer/blob/main/pycytominer/feature_select.py) the single cell plate morphology data per plate.
```
Can you elaborate specifically which feature selection methods?

MattsonCam replied:
Will do!

gwaybio commented on 1.process_data/aggregate_jump_pilot.ipynb:
```diff
@@ -0,0 +1,173 @@
+        "single_cell_plate_path = sys.argv[1]\n",
+        "single_cell_df = pd.read_parquet(single_cell_plate_path)\n",
+        "\n",
+        "plate_name = sys.argv[2]\n",
```
Is this info always going to be redundant with `sys.argv[1]` (plate path)?

MattsonCam replied:
I think you're asking if the first argument ever changes. Yes, the path either points to the feature-selected data or the normalized data

gwaybio commented on 1.process_data/aggregate_sc_data:
```diff
@@ -0,0 +1,32 @@
+#!/bin/bash
```
I recommend naming this file with the `.sh` extension so it's obvious, without viewing its contents, what it is

MattsonCam replied:
The extension is more of a windows thing (my linux professor was adamant about not including file extensions). One of the reasons I include the extension with python files is because my IDE includes the extension by default, otherwise the python files could just use the shebang line. I'll include it for consistency

gwaybio commented on 1.process_data/aggregate_sc_data:
```diff
@@ -0,0 +1,32 @@
+normalized_data_path="${big_drive}/normalized_sc_data"
+feature_selected_data_path="${big_drive}/feature_selected_sc_data"
+
+aggregate() {
```
Can you add some comments (or a docstring) here about what this functionality is doing?

MattsonCam replied:
I'll add those

MattsonCam replied:
Also updated the script docstring

gwaybio commented on 1.process_data/process_data:
```diff
@@ -0,0 +1,34 @@
+#!/bin/bash
```
recommend .sh file name to this file too

gwaybio replied:
oh, I see that this file (process_data.sh) is deleted? Perhaps this is in error?

MattsonCam replied:
That was just how it showed up when I renamed the file. I'll change it back as mentioned in the above comment

gwaybio reviewed (approved):
Comments regarding documentation and file naming improvements. No need for me to review again

Commit:
Updated bash script filenames to include
the filename extensions.
Elaborated on feature selection methods in the README.
Added documentation to the aggregation bash script.
(All according to pr comments)

MattsonCam commented:
Thanks @gwaybio, merging now!
